ext/dom/dom.stub.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+class DOMNode {
+    /** @return DOMNode|false */
+    public function appendChild(DOMNode $newChild) {}
+
+    /** @return string|false */
+    public function C14N(bool $exclusive = false, bool $with_comments = false, ?array $xpath = null, ?array $ns_prefixes = null) {}
+
+    /** @return int|false */
+    public function C14NFile(string $uri, bool $exclusive = false, bool $with_comments = false, ?array $xpath = null, ?array $ns_prefixes = null) {}
+
+    /** @return DOMNode|false */
+    public function cloneNode(bool $recursive = false) {}
+
+    /** @return int */
+    public function getLineNo() {}
+
+    /** @return ?string */
+    public function getNodePath() {}
+
+    /** @return bool */
+    public function hasAttributes() {}
+
+    /** @return bool */
+    public function hasChildNodes() {}
+
+    /** @return DOMNode|false */
+    public function insertBefore(DOMNode $newChild, ?DOMNode $refChild = null) {}
+
+    /** @return bool */
+    public function isDefaultNamespace(string $namespaceURI) {}
+
+    /** @return bool */
+    public function isSameNode(DOMNode $other) {}
+
+    /** @return bool */
+    public function isSupported(string $feature, string $version) {}
+
+    /** @return ?string */
+    public function lookupNamespaceUri(?string $prefix) {}
+
+    /** @return ?string */
+    public function lookupPrefix(string $namespaceURI) {}
+
+    /** @return void */
+    public function normalize() {}
+
+    /** @return DOMNode|false */
+    public function removeChild(DOMNode $oldChild) {}
+
+    /** @return DOMNode|false */
+    public function replaceChild(DOMNode $newChild, DOMNode $oldChild) {}
+}
+
+class DOMAttr {
+    public function __construct(string $name, ?string $value = null) {}
```

Comment:
It works for me with NULL:

```php
var_dump(new DOMAttr("foo", null));
```

Reply:
It gets casted to "", but still allowed to set null.

Reply:
You need to set `strict_types=1`.

Reply:
@beberlei, to actually accept `NULL`, format would have to be `"s|s!"`. Since `value` isn't nullable, it wouldn't make sense to accept `NULL`, though.